web_src/js/features/repo-legacy.js

```diff
@@ -29,6 +29,26 @@ import {hideElem, showElem} from '../utils/dom.js';
 
 const {csrfToken} = window.config;
 
+// if there are draft comments (more than 20 chars), confirm before reloading, to avoid losing comments
+function reloadConfirmDraftComment() {
+  const commentTextareas = [
+    document.querySelector('.edit-content-zone:not(.gt-hidden) textarea'),
+    document.querySelector('.edit_area'),
+  ];
+  for (const textarea of commentTextareas) {
+    // Most users won't feel too sad if they lose a comment with 10 or 20 chars, they can re-type these in seconds.
+    // But if they have typed more (like 50) chars and the comment is lost, they will be very unhappy.
+    if (textarea && textarea.value.trim().length > 20) {
+      textarea.parentElement.scrollIntoView();
+      if (!window.confirm('Page will be reloaded, but there are draft comments. Continuing to reload will discard the comments. Continue?')) {
+        return;
+      }
+      break;
+    }
+  }
+  window.location.reload();
```

Comment:
I'm of the opinion that it's better to just save unsubmitted data into `sessionStorage` and subsequently restore it on page load, but hooking [onbeforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) would also have been an improvement over this method because it works even when the user presses F5 for example.

Reply:
I also proposed so (draft) as one approach.

* https://github.com/go-gitea/gitea/issues/17705#issuecomment-1165105062

Reply:
> but hooking [onbeforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) would also have been an improvement over

Before SimpleMDE/EasyMDE, there is a global `$.areYourSure` confirm (using onbeforeunload) for comment textarea. Everything was just "fine".

However, `$.areYourSure` doesn't work well with SimpleMDE/EasyMDE editor, so the situation became what you see now.

Reply:
I think we should remove this `jquery-are-you-sure` module and implement it on our own to be more flexible.